### PR TITLE
docs: add tutorials for variants

### PR DESCRIPTION
**Pull request by gcroci2**
In this PR:
- Created a notebook for generating missense variant HDF5 files
- Updated the training tutorial notebook and the tutorials' readme for including missense variants
- Updated the files in docs/ and the repo's readme
- Updated the Zenodo links after having uploaded the new data's version
- Improved already existing tutorials and fixed broken links (#462)

Notes
- PR #461 fixed warnings thrown from the training notebook while defining validation and test sets.

### CODE REVIEW

File: tutorials/data_generation_variant.ipynb [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+    "\n",
+    "In this data processing phase, for each variant a local neighborhood around itself is selected according to a radius threshold that the user can customize, and it is mapped to a graph. Nodes either represent residues or atoms, and edges are the interactions between them. Each node and edge can have several different features, which are generated and added during the processing phase as well. Optionally, the graphs can be mapped to volumetric grids (i.e., 3D image-like representations). The mapped data are finally saved into HDF5 files, and can be used for later models' training (for details go to [training_ppi.ipynb](https://github.com/DeepRank/deeprank-core/blob/main/tutorials/training_ppi.ipynb) tutorial). In particular, graphs can be used for the training of Graph Neural Networks (GNNs), and grids can be used for the training of Convolutional Neural Networks (CNNs)."
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:22]:
```suggestion
    "In this data processing phase, a local neighborhood around the mutated residue is selected for each variant according to a radius threshold that the user can customize. All atoms or residues within the threshold are mapped as the nodes to a graph and the interactions between them are the edges of the graph. Each node and edge can have several distinct (structural or physico-chemical) features, which are generated and added during the processing phase as well. Optionally, the graphs can be mapped to volumetric grids (i.e., 3D image-like representations). Finally, the mapped data are saved as HDF5 files, which can be used for training predictive models (for details see [training_ppi.ipynb](https://github.com/DeepRank/deeprank-core/blob/main/tutorials/training_ppi.ipynb) tutorial). In particular, graphs can be used for the training of Graph Neural Networks (GNNs), and grids can be used for the training of Convolutional Neural Networks (CNNs)."
```

I think this flows a bit better and is a bit clearer than how it was phrased before. If you like it like this, also change the ppi tutorial accordingly.

File: tutorials/data_generation_variant.ipynb [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Input Data\n",
+    "\n",
+    "The example data used in this tutorial are available on Zenodo at [this record address](https://zenodo.org/record/8187806). To download the raw data used in this tutorial, please visit the link and download `data_raw.zip`. Unzip it, and save the `data_raw/` folder in the same directory as this notebook. The name and the location of the folder are optional but recommended, as they are the name and the location we will use to refer to the folder throughout the tutorial.\n",
+    "\n",
+    "Note that the dataset contains only 96 data points, which is not enough to develop an impactful predictive model, and the scope of its use is indeed only demonstrative and informative for the users."
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:34]: I think my download only contains 58 pdb files, but the csv has 100 entries. Not sure how those relate, but they are both different from 96 :)
I mentioned 58 a few times in other comments. If 100 is accurate, ignore/adjust what I said there ;)

File: tutorials/data_generation_ppi.ipynb [hunk 3]
```diff
@@ -164,7 +164,7 @@
     "\n",
     "- A `.pdb` file, representing the protein-protein structural complex.\n",
     "- The ids of the two chains composing the complex. In our use case, \"M\" indicates the protein's chain and \"P\" the peptide's chain.\n",
```
Comment (DaniBodor) [inline tutorials/data_generation_ppi.ipynb:166]:
```suggestion
    "- The ids of the two chains composing the complex. In our use case, \"M\" indicates the MHC protein chain and \"P\" the peptide chain.\n",
```

File: tutorials/data_generation_variant.ipynb [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+    "\n",
+    "- In a `SingleResidueVariantResidueQuery` each node represents one amino acid residue.\n",
+    "- In a `SingleResidueVariantAtomicQuery` each node represents one atom within the amino acid residues.\n",
+    "\n",
+    "A query takes as inputs:\n",
+    "\n",
+    "- A `.pdb` file, representing the protein structure containing the missense variant.\n",
+    "- The chain identifier of the variant residue.\n",
+    "- The positional number of the variant residue.\n",
+    "- The insertion code, ??\n",
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:187]: ask Gayatri or Li what this is?
it is also never tested (as not None), so maybe we should include a test for it at some point when we know what it is. We can then open a separate issue for that.

File: tutorials/data_generation_variant.ipynb [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+    "For each missense variant, so for each data point, a query can be created and added to the `QueryCollection` object, to be processed later on. Different types of queries exist, based on the molecular resolution needed:\n",
+    "\n",
+    "- In a `SingleResidueVariantResidueQuery` each node represents one amino acid residue.\n",
+    "- In a `SingleResidueVariantAtomicQuery` each node represents one atom within the amino acid residues.\n",
+    "\n",
+    "A query takes as inputs:\n",
+    "\n",
+    "- A `.pdb` file, representing the protein structure containing the missense variant.\n",
+    "- The chain identifier of the variant residue.\n",
+    "- The positional number of the variant residue.\n",
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:186]:
```suggestion
    "- The chain id of the variant residue.\n",
    "- The residue number of the missense mutation.\n",
```

(we call it id in the ppi tutorial as well)

File: tutorials/data_generation_variant.ipynb [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+    "- The insertion code, ??\n",
+    "- The wildtype amino acid. \n",
+    "- The variant amino acid. \n",
+    "- The radius, according to which determine the residues/atoms that will be included in the graph, in Ångström.\n",
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:190]:
```suggestion
    "- The radius, which determines the threshold distance (in Ångström) for residues/atoms surrounding the mutation that will be included in the graph.\n",
```

File: tutorials/data_generation_variant.ipynb [hunk 7]
```diff
@@ -0,0 +1,572 @@
...
+    "- In a `SingleResidueVariantAtomicQuery` each node represents one atom within the amino acid residues.\n",
+    "\n",
+    "A query takes as inputs:\n",
+    "\n",
+    "- A `.pdb` file, representing the protein structure containing the missense variant.\n",
+    "- The chain identifier of the variant residue.\n",
+    "- The positional number of the variant residue.\n",
+    "- The insertion code, ??\n",
+    "- The wildtype amino acid. \n",
+    "- The variant amino acid. \n",
```
Comment (DaniBodor) [inline tutorials/data_generation_variant.ipynb:189]: Not really part of this PR, but something to consider: why are we asking the position and the wt amino acid as separate inputs?
In principle, the position should be sufficient, because that will already determine the wt aa. If we are doing it to as a backup in case the user makes a mistake here, then I'd argue requesting the wt amino acid (or the full mutation) in the default notation for this (e.g. Ser123 or S123 for the residue / Ser123Thr or S123T for a mutation).
Anyway, if we want to change it, that should be a separate issue.

File: tutorials/data_generation_ppi.ipynb [hunk 8]
```diff
@@ -239,6 +239,7 @@
     "queries.process(\n",
     "\tprefix = os.path.join(processed_data_path, \"residue\", \"proc\"),\n",
     "\tfeature_modules = [components, contact],\n",
+    "    cpu_count = 8,\n",
```
Comment (DaniBodor) [inline tutorials/data_generation_ppi.ipynb:242]: why did you decide to hard code this?

**Review by DaniBodor — CHANGES_REQUESTED** [review]
I haven't managed to copy the raw data into my wsl environment of my new laptop, so I wasn't able to test run the notebook. Something about permissions, which I have been having difficulties with for a few different things.
So far, I have only looked at the markdown cells and glossed over the code. If you want me to test run the notebook as well, let me know and I will try to see how to move the data to wsl.

File: README.md [hunk 9]
```diff
@@ -13,20 +13,19 @@
...
-- Predefined atom-level and residue-level PPI feature types
+- Predefined atom-level and residue-level feature types
   - e.g. atomic density, vdw energy, residue contacts, PSSM, etc.
-- Predefined target type
+- Predefined target types
   - e.g. binary class, CAPRI categories, DockQ, RMSD, FNAT, etc.
 - Flexible definition of both new features and targets
-- Graphs and grids feature mapping
+- Graphs and grids features mapping
 - Efficient data storage in HDF5 format
```
[the remark this answers is not in the document]
Comment (gcroci2) [reply README.md:27]: It is efficient! The logic can be improved to make it much more efficient, but still

File: tutorials/training.ipynb [hunk 10]
```diff
@@ -31,11 +31,11 @@
...
+    "Alternatively, preprocessed HDF5 files can be downloaded directly from Zenodo at [this record address](https://zenodo.org/record/8187806). To download the data used in this tutorial, please visit the link and download `data_processed.zip`. Unzip it, and save the `data_processed/` folder in the same directory as this notebook. The name and the location of the folder are optional but recommended, as they are the name and the location we will use to refer to the folder throughout the tutorial. \n",
     "\n",
-    "Note that the dataset contains only 100 data points, which is not enough to develop an impactful predictive model, and the scope of its use is indeed only demonstrative and informative for the users."
+    "Note that the datasets contain only ~100 data points each, which is not enough to develop an impactful predictive model, and the scope of their use is indeed only demonstrative and informative for the users."
```
[the remark this answers is not in the document]
Comment (gcroci2) [reply tutorials/training.ipynb:38]: the data points are around 100 for both examples (58 PDB files for variants that contain multiple variants)

File: tutorials/data_generation_variant.ipynb (hunk 4 above)
Comment (gcroci2) [reply tutorials/data_generation_variant.ipynb:187]: The insertion code is used when two residues have the same numbering. The combination of residue numbering and insertion code defines the unique residue (https://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM)

File: tutorials/data_generation_ppi.ipynb (hunk 8 above)
Comment (gcroci2) [reply tutorials/data_generation_ppi.ipynb:242]: Because depending on the number of cpus of the machine of the user, this can get out of control (I run it on Snellius without setting it and it was generating more than 50 files). Setting it to 8 (or whatever small number) ensures that this is the maximum number of files generated. If the user has fewer cpus available, then it generates fewer files but it's fine.

File: README.md [hunk 11]
```diff
@@ -13,20 +13,19 @@
...
-DeeprankCore contains useful APIs for pre-processing PPI data, computing features and targets, as well as training and testing GNN and CNN models.
+DeepRank-Core allows to transform and store 3D representations of both PPIs and individual proteins' variants into grids or graphs containing structural and physico-chemical information, which can then be used for training neural networks for whatever specific pattern of interest for the user. DeepRank-Core also offers a pre-implemented training pipeline which can use either CNNs or GNNs, as well as handy output exporters for evaluating performances. 
 
 Main features:
-- Predefined atom-level and residue-level PPI feature types
+- Predefined atom-level and residue-level feature types
   - e.g. atomic density, vdw energy, residue contacts, PSSM, etc.
-- Predefined target type
+- Predefined target types
   - e.g. binary class, CAPRI categories, DockQ, RMSD, FNAT, etc.
```
[the remark this answers is not in the document]
Comment (gcroci2) [reply README.md:24]: I removed the target BA from everywhere (I explicitly hardcoded in the tests when needed). The file target_storage.py should indeed contain only the targets that can be computed automatically from the package, defined in `compute_targets` function in targets.py.

File: README.md [hunk 12]
```diff
@@ -13,20 +13,19 @@
...
 - Flexible definition of both new features and targets
-- Graphs and grids feature mapping
+- Graphs and grids features mapping
```
[the remark this answers is not in the document]
Comment (gcroci2) [reply README.md:26]: This comes from the previous deeprank-gnn, I guess it means that we map the features to the graphs/grids

File: tutorials/data_generation_variant.ipynb (hunk 7 above)
Comment (gcroci2) [reply tutorials/data_generation_variant.ipynb:189]: Reflecting more on this, if you know the position you don't know the wt aa, but only the variant (because you have the pdb file with the variant only, and not with the original aa). Or am I wrong?

File: deeprankcore.png
[the remark this answers is not in the document]
Comment (DaniBodor) [reply deeprankcore.png:1]: wow. I really like the new image!

File: tutorials/images/data_generation_variants.png
[the remark this answers is not in the document]
Comment (DaniBodor) [reply tutorials/images/data_generation_variants.png:1]: tiny comments (non essential):
- maybe use the same "grid" cube as you did for deeprankcore.png.
- capitalize each word to be consistent with deeprankcore.png

File: tutorials/data_generation_variant.ipynb (hunk 2 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb:34]: I misunderstood that one pdb can have multiple variants, so I think no change needed.
If the csv has some excess unused entries we can just leave it as is, imo.

File: tutorials/data_generation_variant.ipynb (hunk 4 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb]: The tutorial still shows a couple of question marks here. This should either be replaced by above comment or should be removed as it looks kind of iffy.

File: tutorials/data_generation_variant.ipynb (hunk 5 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb]: unchanged?

File: tutorials/data_generation_variant.ipynb (hunk 6 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb]: also unchanged?

File: tutorials/data_generation_variant.ipynb (hunk 7 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb:189]: But the code can read the wt aa from just the residue number.

File: README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (DaniBodor) [reply README.md]: I would actually imagine that binding affinity is a pretty generic target type for protein protein interactions.

**Review by DaniBodor — APPROVED** [review]
I think it's all fine. Minor suggestions.
The one thing that really confuses me is that you removed BA as a default target (see comment below).

File: tutorials/data_generation_variant.ipynb (hunk 4 above)
Comment (gcroci2) [reply tutorials/data_generation_variant.ipynb]: Look at the updated commits, I edited it

File: tutorials/data_generation_variant.ipynb (hunk 7 above)
Comment (gcroci2) [reply tutorials/data_generation_variant.ipynb:189]: from my understanding this is not true, because the PDB file contains the variant aa only. So if the code looks at the residue number it finds the variant aa only, and has no info about the original wt (since it's not in the pdb). But maybe the wt aa is in the pdb as well, next to the variant aa?

Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb:189]: I opened a separate issue on this discussion #470

File: tutorials/data_generation_variant.ipynb (hunk 4 above)
Comment (DaniBodor) [reply tutorials/data_generation_variant.ipynb]: sorry, somehow it wasnt up to date

**Review by DaniBodor — APPROVED** [review]
